src/Camera.cpp

```diff
@@ -74,6 +74,33 @@ void Camera::changeBeta(float deltaBeta) {
    player->direction = glm::rotateY(glm::vec3(1.0f, 0.0f, 0.0f), -betaRad);
 }
 
+/* Calculates the angle of the given direction and the original direction vector
+ * to determine the rotation angle to apply to the current direction vector.
+ * Note: The dot product doesn't specify sign therefore if the z component
+ * of the given vector is negative the angle will be negated.
+ * Note: If the player is in reverse the base direction for the angle
+ * calculation is reversed.
+ */
+void Camera::calcCamAndPlayerOrient(glm::vec3 dirOrigOrient, bool reverse) {
+   glm::vec3 origDirection, unitDir = glm::normalize(dirOrigOrient);

```

Comment:
Minor nit, but can you separate this into two lines? Tripped me up while reading it.